### PR TITLE
bitCount support FixedString data type

### PR DESCRIPTION
**Pull request by ucasfl**
### Changelog category (leave one):

- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
`bitCount` function support `FixedString` and `String` data type.

### CODE REVIEW

**Comment by ucasfl** [comment]
@alexey-milovidov Not sure is there any better way to implement this.

**Comment by ucasfl** [comment]
> It looks rather good 👍 The iteration by bytes might be suboptimal, but I hope the compiler will figure out how to optimize.
> 
> Let's also add a test with a bunch of random strings and compare for ground truth:
> 
> ```
> length(replaceAll(bin(toFixedString(string, 16)), '0', ''))
> ```
> 
> The return type UInt64 looks too large. Can we determine it based on the size of FixedString?
> 
> Can we also implement it for the String type?

OK.

File: src/Functions/FunctionUnaryArithmetic.h [hunk 1]
```diff
@@ -278,16 +348,80 @@ class FunctionUnaryArithmetic : public IFunction
...
+
+                            auto col_res = ColumnUInt64::create(size);
+                            auto & vec_res = col_res->getData();
+
+                            const auto & chars = col->getChars();
+                            const auto & offsets = col->getOffsets();
+                            for (size_t i = 0; i < size; ++i)
+                            {
+                                vec_res[i] = FixedStringUnaryOperationReduceImpl<Op<UInt8>>::vector(
+                                    chars.data() + offsets[i - 1], chars.data() + offsets[i]);
```
Comment (alexey-milovidov) [inline src/Functions/FunctionUnaryArithmetic.h:403]: Here we are processing one extra zero byte for nothing.

File: src/Functions/FunctionUnaryArithmetic.h [hunk 2]
```diff
@@ -130,6 +132,46 @@ struct FixedStringUnaryOperationImpl
     }
 };
 
+template <typename Op>
+struct FixedStringUnaryOperationReduceImpl
```
Comment (alexey-milovidov) [inline src/Functions/FunctionUnaryArithmetic.h:136]: Now it should be renamed to `StringUnaryOperationReduceImpl`, as nothing "Fixed" remains.

File: src/Functions/FunctionUnaryArithmetic.h [hunk 3]
```diff
@@ -130,6 +132,46 @@ struct FixedStringUnaryOperationImpl
...
+{
+    MULTITARGET_FUNCTION_AVX512BW_AVX512F_AVX2_SSE42(
+        MULTITARGET_FUNCTION_HEADER(static UInt64 NO_INLINE),
+        vectorImpl,
+        MULTITARGET_FUNCTION_BODY((const UInt8 * start, const UInt8 * end) { /// NOLINT
```
Comment (alexey-milovidov) [inline src/Functions/FunctionUnaryArithmetic.h:141]: I think if we move `{` to a new line, no NOLINT will be needed.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**